src/wasm.rs

```diff
@@ -62,6 +64,72 @@ impl Drop for WasmStreaming {
   }
 }
 
+impl WasmModuleObject {
+  /**
+   * Efficiently re-create a WasmModuleObject, without recompiling, from
+   * a CompiledWasmModule.
+   */
+  pub fn from_compiled_module<'s>(
+    scope: &mut HandleScope<'s>,
+    compiled_module: &CompiledWasmModule,
+  ) -> Option<Local<'s, WasmModuleObject>> {
+    unsafe {
+      scope.cast_local(|sd| {
+        v8__WasmModuleObject__FromCompiledModule(
+          sd.get_isolate_ptr(),
+          compiled_module.0,
+        )
+      })
+    }
+  }
+
+  /**
+   * Get the compiled module for this module object. The compiled module can be
+   * shared by several module objects.
+   */
+  pub fn get_compiled_module(&self) -> CompiledWasmModule {
+    let ptr = unsafe { v8__WasmModuleObject__GetCompiledModule(self) };
+    CompiledWasmModule(ptr)
+  }
+}
+
+// Type-erased v8::CompiledWasmModule. We need this because the C++
+// v8::CompiledWasmModule must be destructed because its private fields hold
+// pointers that must be freed, but v8::CompiledWasmModule itself doesn't have
+// a destructor. Therefore, in order to avoid memory leaks, the Rust-side
+// CompiledWasmModule must be a pointer to a C++ allocation of
+// v8::CompiledWasmModule.
+#[repr(C)]
+struct InternalCompiledWasmModule(Opaque);
+
+/// Wrapper around a compiled WebAssembly module, which is potentially shared by
+/// different WasmModuleObjects.
+pub struct CompiledWasmModule(*mut InternalCompiledWasmModule);
+
+impl CompiledWasmModule {
+  /**
+   * Get the (wasm-encoded) wire bytes that were used to compile this module.
+   */
+  pub fn get_wire_bytes_ref(&self) -> &[u8] {
```

Comment:
Use triple slash comment here

Reply:
My bad, I was being lazy and just directly copied those from `v8.h` 😅